tests/test_is_valid_email.py

```diff
@@ -1,121 +1,24 @@
 
-def test_concatenated_emails():
-    """
-    Test the is_valid_email function with two valid emails concatenated.
-    This should not be considered a valid email.
-    """
-    email = "test.user1@inspection.gc.ca" + "test.user2@inspection.gc.ca"
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
 
-def test_email_with_unsupported_symbols():
-    """
-    Test the is_valid_email function with an email containing unsupported symbols.
-    Such an email should not be considered valid.
-    """
-    email = "test.user!#%^&*()+=[]{}|;<>?@inspection.gc.ca"
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
-def test_email_with_lookalike_utf8_characters():
-    """
-    Test the is_valid_email function with an email containing look-alike UTF-8 characters.
-    Such an email should not be considered valid as it's potentially misleading.
-    """
-    # Using Cyrillic 'а' which looks like Latin 'a'
-    email = "test.userа@inspection.gc.ca"
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
-def test_misspelled_inspection_subdomain():
-    """
-    Test the is_valid_email function with a misspelled "inspection" subdomain.
-    """
-    email = "user@inspektion.gc.ca"
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
-def test_extra_characters_in_canada_domain():
-    """
-    Test the is_valid_email function with an extra character in the "canada" domain.
-    """
-    email = "user@canadaa.ca"
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
-def test_missing_domain_extension():
-    """
-    Test the is_valid_email function with a missing domain extension.
-    """
-    email = "user@canada."
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
-def test_missing_subdomain():
-    """
-    Test the is_valid_email function with a missing subdomain.
-    """
-    email = "user@.gc.ca"
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
-def test_extra_characters_in_tld():
-    """
-    Test the is_valid_email function with extra characters in the TLD.
-    """
-    email = "user@inspection.gc.caa"
-    try:
-        is_valid_email(email)
-        assert False, f"Expected {email} to be invalid but was not."
-    except EmailError:
-        assert True
+def test_invalid_emails(invalid_emails: list[str]):
```

Comment:
as discussed we need the test suite to be readable with the tested value and test name clearly readable.

Reply:
done